# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/layers/ipc/CompositorBridgeParent.h
# Commit: eac17a862c29
# Full Hash: eac17a862c29a52ae1a389c5e55c0a1cec0db7b6
# Author: Andrew Osmond <aosmond@mozilla.com>
# Date: 2020-01-11 09:48:29
# Regressor Bug: 1600472
# File Overlap Count: 2
# Description:
#   Bug 1600472 - Disable allowing sacrificing subpixel anti-aliasing for small screens. r=jrmuizel
#   
#   This patch only allows sacrificing subpixel anti-aliasing when the
#   screen size is larger than WUXGA, and when the force disable pref is not
#   set. In the future, we may also add disable this for high end GPUs.
# ==============================================================================

diff -r b6e5faa0693d -r eac17a862c29 gfx/layers/ipc/CompositorBridgeParent.h
--- a/gfx/layers/ipc/CompositorBridgeParent.h	Fri Jan 10 18:24:13 2020 +0000
+++ b/gfx/layers/ipc/CompositorBridgeParent.h	Fri Jan 10 16:57:55 2020 +0000
@@ -527,6 +527,11 @@
   void InvalidateRemoteLayers();
 
   /**
+   * Initialize statics.
+   */
+  static void InitializeStatics();
+
+  /**
    * Returns a pointer to the CompositorBridgeParent corresponding to the given
    * ID.
    */
@@ -718,6 +723,16 @@
   static void DeallocateLayerTreeId(LayersId aId);
 
   /**
+   * Notify the compositor the quality settings have been updated.
+   */
+  static void UpdateQualitySettings();
+
+  /**
+   * Notify the compositor the debug flags have been updated.
+   */
+  static void UpdateDebugFlags();
+
+  /**
    * Wrap the data structure to be sent over IPC.
    */
   Maybe<CollectedFramesParams> WrapCollectedFrames(CollectedFrames&& aFrames);
@@ -797,6 +812,11 @@
   template <typename Lambda>
   inline void ForEachIndirectLayerTree(const Lambda& aCallback);
 
+  // The indirect layer tree lock must be held before calling this function.
+  // Callback should take (LayerTreeState* aState, const LayersId& aLayersId)
+  template <typename Lambda>
+  static inline void ForEachWebRenderBridgeParent(const Lambda& aCallback);
+
   RefPtr<HostLayerManager> mLayerManager;
   RefPtr<Compositor> mCompositor;
   RefPtr<AsyncCompositionManager> mCompositionManager;